### PR TITLE
Add table oci_dns_zone. Closes #98

### LINKED ISSUE #98: Add table oci_dns_zone
**References**
https://github.com/oracle/oci-go-sdk/blob/master/dns/zone.go

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/oci_dns_zone []

PRETEST: tests/oci_dns_zone

TEST: tests/oci_dns_zone
Running terraform
oci_dns_zone.named_test_resource: Creating...
oci_dns_zone.named_test_resource: Still creating... [10s elapsed]
oci_dns_zone.named_test_resource: Creation complete after 12s [id=ocid1.dns-zone.oc1..8274e69f82d742dc84d234a8da7ddf97]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

resource_id = "ocid1.dns-zone.oc1..8274e69f82d742dc84d234a8da7ddf97"
resource_name = "turbot.com"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "id": "ocid1.dns-zone.oc1..8274e69f82d742dc84d234a8da7ddf97",
    "lifecycle_state": "ACTIVE",
    "name": "turbot.com"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "id": "ocid1.dns-zone.oc1..8274e69f82d742dc84d234a8da7ddf97",
    "lifecycle_state": "ACTIVE",
    "name": "turbot.com"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "turbot.com"
  }
]
✔ PASSED

POSTTEST: tests/oci_dns_zone

TEARDOWN: tests/oci_dns_zone

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  id,
  name,
  lifecycle_state,
  time_created
from
  oci_dns_zone;
+------------------------------------------------------+------------+-----------------+---------------------+
| id                                                   | name       | lifecycle_state | time_created        |
+------------------------------------------------------+------------+-----------------+---------------------+
| ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 | turbot.com | ACTIVE          | 2021-04-20 07:35:51 |
+------------------------------------------------------+------------+-----------------+---------------------+

select
  name,
  id,
  lifecycle_state
from
  oci_dns_zone
where
  lifecycle_state = 'ACTIVE';
+------------+------------------------------------------------------+-----------------+
| name       | id                                                   | lifecycle_state |
+------------+------------------------------------------------------+-----------------+
| turbot.com | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 | ACTIVE          |
+------------+------------------------------------------------------+-----------------+
```
</details>

### CODE REVIEW

File: oci-test/tests/oci_dns_zone/variables.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+resource "oci_dns_zone" "named_test_resource" {
+    #Required
+    compartment_id = var.tenancy_ocid
+    name = "turbot.com"
+    zone_type = "PRIMARY"
+}
+
+output "resource_name" {
+  value = "turbot.com"
```
Comment (khushboo9024) [inline oci-test/tests/oci_dns_zone/variables.tf:38]:
```suggestion
  value = "oci_dns_zone. named_test_resource. name"
```

File: oci/service.go [hunk 2]
```diff
@@ -230,6 +232,44 @@ func coreVirtualNetworkService(ctx context.Context, d *plugin.QueryData, region
 	return sess, nil
 }
 
+// DNSService returns the service client for OCI DNS Service
+func DNSService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
```
Comment (khushboo9024) [inline oci/service.go:236]: Is there any specific reason we are keeping `dns` in caps ?

File: oci/table_oci_dns_zone.go [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+//// HYDRATE FUNCTIONS
+
+func getDnsZone(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("oci.getDnsZone", "Compartment", compartment, "OCI_REGION", region)
+
+	// Rstrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
```
Comment (khushboo9024) [inline oci/table_oci_dns_zone.go:205]:
```> select name, id from oci_new.oci_dns_zone
+---------------+------------------------------------------------------+
| name          | id                                                   |
+---------------+------------------------------------------------------+
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |
+---------------+------------------------------------------------------+
> select name, id, region  from oci_new.oci_dns_zone
+---------------+------------------------------------------------------+--------+
| name          | id                                                   | region |
+---------------+------------------------------------------------------+--------+
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |        |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |        |
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |        |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |        |
+---------------+------------------------------------------------------+--------+
> select * from oci_new.oci_dns_zone where id = 'ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52'
Error: get call returned 2 results - the key column is not globally unique
> 

```
we are getting duplicate data in list call, get call is also not working and region column is always empty.
Seems like  this resource is same a cloudguard  service, but not sure, please check on this.

File: oci/table_oci_dns_zone.go [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+			},
+
+			// other columns
+			{
+				Name:        "scope",
+				Description: "The scope of the zone.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "is_protected",
```
Comment (khushboo9024) [inline oci/table_oci_dns_zone.go:65]: `is_protected` column should be above `scope` column.

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Please make the suggested changes.

File: oci/service.go (hunk 2 above)
Comment (bigdatasourav) [reply oci/service.go:236]: no reason, updated .

File: oci/table_oci_dns_zone.go (hunk 3 above)
Comment (bigdatasourav) [reply oci/table_oci_dns_zone.go:203]: updated getmatrix, removed region.

**Review by khushboo9024 — APPROVED** [review]
LGTM

File: docs/tables/oci_dns_zone.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+select
+  id,
+  name,
+  lifecycle_state,
+  time_created
+from
+  oci_dns_zone;
+```
+
+### List active DNS zones
```
Comment (rajmohanty17) [inline docs/tables/oci_dns_zone.md:19]: Make this as `List inactive DNS zones`

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**